packages/core/src/logger/defaultLogger.ts

```diff
@@ -0,0 +1,25 @@
+import { Level, LoggerInterface } from '@apimatic/core-interfaces';
+
+export class ConsoleLogger implements LoggerInterface {
+  public log(level: Level, message: string, params: Record<string, any>): void {
+    const formattedMessage = formatMessage(message, params);
+    // tslint:disable-next-line:no-console
+    console.log(level + ': ' + formattedMessage);
+  }
+}
+
+const formatMessage = (msg: string, obj: Record<string, any>): string => {
+  let formattedMsg = msg;
+  for (const [key, value] of Object.entries(obj)) {
+    if (typeof value === 'object') {
+      // If the value is an object, stringify it
+      const placeholder = '${' + key + '}';
+      formattedMsg = formattedMsg.replace(placeholder, JSON.stringify(value));
+    } else {
+      // Replace placeholders with corresponding values
+      const placeholder = '${' + key + '}';
+      formattedMsg = formattedMsg.replace(placeholder, String(value));
+    }
+  }
```

Comment:
This logic has a security vulnerability. The string interpolation logic you have implemented replaces tokens/placeholders one by one, updating the string template as you go. However, an interpolated value might also contain a placeholder-like syntax i.e. `${something}` leading to an unintended interpolation happening in the next iteration.

Reply:
I have replaced the logic using the regex operation to remove this vulnerability